pkg/provider/gcp/secretmanager/secretsmanager.go

```diff
@@ -0,0 +1,130 @@
+	"github.com/external-secrets/external-secrets/pkg/provider"
+	"github.com/external-secrets/external-secrets/pkg/provider/schema"
+)
+
+const (
+	cloudPlatformRole = "https://www.googleapis.com/auth/cloud-platform"
+	defaultVersion    = "latest"
+)
+
+type GoogleSecretManagerClient interface {
+	AccessSecretVersion(ctx context.Context, req *secretmanagerpb.AccessSecretVersionRequest, opts ...gax.CallOption) (*secretmanagerpb.AccessSecretVersionResponse, error)
+	Close() error
+}
+
+// GCP is a provider for GCP Secret Manager.
+type GCP struct {
+	projectID           string
+	SecretManagerClient GoogleSecretManagerClient
+}
+
+// NewClient constructs a GCP Provider.
+func (sm *GCP) NewClient(ctx context.Context, store esv1alpha1.GenericStore, kube client.Client, namespace string) (provider.SecretsClient, error) {
+	// Fetch credential Secret
+	credentialsSecret := &corev1.Secret{}
+	credentialsSecretName := store.GetSpec().Provider.GCPSM.Auth.SecretRef.SecretAccessKey.Name
+	objectKey := types.NamespacedName{Name: credentialsSecretName, Namespace: store.GetNamespace()}
+	err := kube.Get(ctx, objectKey, credentialsSecret)
+	if err != nil {
+		log.Panicf(err.Error(), "Failed to get credentials Secret")
+		return nil, err
+	}
+
+	credentials := credentialsSecret.Data[store.GetSpec().Provider.GCPSM.Auth.SecretRef.SecretAccessKey.Key]
+	if len(credentials) == 0 {
+		return nil, fmt.Errorf("credentials GCP invalid/not provided")
+	}
+
+	projectID := store.GetSpec().Provider.GCPSM.ProjectID
+
+	sm.projectID = projectID
+
+	config, err := google.JWTConfigFromJSON(credentials, cloudPlatformRole)
+	if err != nil {
+		log.Panicf(err.Error(), "Failed to process the provided JSON credentials")
+		return nil, err
+	}
+
+	ts := config.TokenSource(ctx)
+
+	client, err := secretmanager.NewClient(ctx, option.WithTokenSource(ts))
+	if err != nil {
+		return nil, fmt.Errorf("failed to create GCP secretmanager client: %w", err)
+	}
+	sm.SecretManagerClient = client
+	return sm, nil
+}
+
+// GetSecret returns a single secret from the provider.
+func (sm *GCP) GetSecret(ctx context.Context, ref esv1alpha1.ExternalSecretDataRemoteRef) ([]byte, error) {
+	if sm.SecretManagerClient == nil || sm.projectID == "" {
+		return nil, fmt.Errorf("provider GCP is not initialized")
+	}
+
+	version := ref.Version
+	if version == "" {
+		version = defaultVersion
+	}
+
+	resourceName := fmt.Sprintf("projects/%s/secrets/%s/versions/%s", sm.projectID, ref.Key, version)
+
+	req := &secretmanagerpb.AccessSecretVersionRequest{
+		Name: resourceName,
+	}
+
+	result, err := sm.SecretManagerClient.AccessSecretVersion(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+	err = sm.SecretManagerClient.Close()
+	if err != nil {
+		return nil, err
+	}
+	return []byte(string(result.Payload.Data)), nil
+}
+
+// GetSecretMap returns multiple k/v pairs from the provider.
+func (sm *GCP) GetSecretMap(ctx context.Context, ref esv1alpha1.ExternalSecretDataRemoteRef) (map[string][]byte, error) {
```

Comment:
Is the `GetSecretMap` not possible today with GCP SM or will this still be implemented?